dash/_callback.py

```diff
@@ -440,10 +458,19 @@ def add_context(*args, **kwargs):
                     isinstance(output_value, dict)
                     and "long_callback_error" in output_value
                 ):
-                    error = output_value.get("long_callback_error")
-                    raise LongCallbackError(
+                    error = output_value.get("long_callback_error", {})
+                    exc = LongCallbackError(
                         f"An error occurred inside a long callback: {error['msg']}\n{error['tb']}"
                     )
+                    if error_handler:
+                        error_handler(exc)
+                        output_value = NoUpdate()
```

Comment:
see https://github.com/plotly/dash/pull/2903/files#r1669336763

```suggestion
                        output_value = error_handler(exc)
                        output_value = output_value or NoUpdate()
```